File: 6. Zigzag Conversion/code.cpp
class Solution {
    public String convert(String s, int numRows) {
        
        String str="";
       if(numRows==1 || (s.length()/numRows)==0) return s;
        
            for(int j=0;j<s.length();j+=(2*numRows-2))
                str+=s.charAt(j);
        boolean x=false;
        for(int i=1;i<numRows-1;i++){
            x=false;
            for(int j=i;j<s.length();){
                
                str+=s.charAt(j);
                if(!x)
                {
                    j+=2+(2*(numRows-i-2));
                    x=true;
                    }
                else
                {
                    j+=2+(2*(i-1));
                   x=false; 
                }
            }
                
        }
         for(int j=numRows-1;j<s.length();j+=(2*numRows-2))
                str+=s.charAt(j);
        return str;
    }
}

File: 6. Zigzag Conversion/code.java
class Solution {
    public String convert(String s, int n) {
        
        String s1[] = new String[n];
        for(int i=0;i<n;i++) s1[i]="";
        int i=0;
        while(i<s.length()){
            for(int j=0;j<n && i<s.length();j++){
                s1[j]+=s.charAt(i++);
            }
            
            for(int j=n-2;j>=1 && i<s.length();j--){
                s1[j]+=s.charAt(i++);
            }
        }
        
        String ans="";
        
        for(i=0;i<n;i++){
            ans+=s1[i];
            // System.out.println(s1[i]);
        }
        
        return ans;
    }
}